Catappult/appcoins-wallet-android
# [task/APPC-2802] Promo code message with App Name

**What does this PR do?**

   Adds a new message to when a user enters a promo code with bonus only for a specific app. Also improves and refactors parts of the code.

**Database changed?**

   No

**How should this be manually tested?**

  Test the promo code feature and insert a code with and without a specific app bonus to see the different messages.

**What are the relevant tickets?**

  [APPC-2802](https://aptoide.atlassian.net/browse/APPC-2802)

**Code Review Checklist**
